samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporter.scala

```diff
@@ -112,6 +116,8 @@ class MetricsSnapshotReporter(
         registry.getGroup(group).asScala.foreach {
           case (name, metric) =>
             metric.visit(new MetricsVisitor {
+              // for listGauge the value is returned as a list, which gets serialized
```

Comment:
I think you might need to update AmfReporter.java too.

Reply:
Sure, will do. 
The plan was also to use AmfReporter's filter to filter out the "exception" metric, because amf cant handle exceptions/string types